src/OpenSage.Game/Terrain/Bridge.cs

```diff
@@ -18,36 +19,30 @@ public sealed class Bridge : DisposableBase
         private readonly Model _model;
         private readonly ModelInstance _modelInstance;
         private readonly List<Tuple<ModelSubObject, Matrix4x4>> _meshes;
-        private readonly List<GameObject> _towers;
 
-        private Bridge(
+        internal Bridge(
             ContentManager contentManager,
             HeightMap heightMap,
-            BridgeTemplate template,
+            MapObject mapObject,
             in Vector3 startPosition,
             in Vector3 endPosition,
-            Model model,
             GameObjectCollection parent)
         {
-            _model = model;
-
-            _modelInstance = AddDisposable(_model.CreateInstance(contentManager));
-
-            _modelInstance.Update(TimeInterval.Zero);
+            var template = contentManager.IniDataContext.FindBridgeTemplate(mapObject.TypeName);
 
-            _towers = new List<GameObject>();
+            var modelPath = Path.Combine("Art", "W3D", template.BridgeModelName + ".W3D");
```

Comment:
This won't work for >=BFME since the modelpath is constructed differently there (models are in a subfolder named with the first two characters of the modelname)